h_malloc.c

```diff
@@ -1368,6 +1477,11 @@ EXPORT void *h_calloc(size_t nmemb, size_t size) {
     if (!ZERO_ON_FREE && likely(p != NULL) && total_size && total_size <= max_slab_size_class) {
         memset(p, 0, total_size - canary_size);
     }
+#ifdef HAS_ARM_MTE
+    // use an assert instead of adding a conditional to memset() above (freed memory is always
+    // zeroed when MTE is enabled)
+    static_assert(ZERO_ON_FREE, "disabling ZERO_ON_FREE reduces performance when ARM MTE is enabled");
```

Comment:
This statement is a bit counter-intuitive, it would be nice to explain why its is true.

Reply:
Comment refers to memset() above (line 1478). If zero-on-free would be disabled, that memset() would be executed unnecessarily.

Reply:
Oh, now I understand!